sdk/monitor/opentelemetry-exporter/src/platform/nodejs/persist/fileSystemPersist.ts

```diff
@@ -28,120 +35,99 @@ export class FileSystemPersist implements PersistentStorage {
-    this._getFirstFileOnDisk((error, buffer) => {
-      if (error) {
-        cb(error);
-      } else if (buffer) {
-        this._logger.info("Found a filesystem persisted file");
-        cb(null, JSON.parse(buffer.toString("utf8")));
-      } else {
-        cb(null);
-      }
-    });
+    const buffer = await this._getFirstFileOnDisk();
+    if (buffer) {
+      return JSON.parse(buffer.toString("utf8"));
+    }
+    return null;
   }
 
   /**
    * Check for temp telemetry files
    * reads the first file if exist, deletes it and tries to send its load
    */
-  private _getFirstFileOnDisk(callback: (error: Error | null, value?: Buffer) => void): void {
+  private async _getFirstFileOnDisk(): Promise<Buffer | null> {
     const tempDir = path.join(
       os.tmpdir(),
       FileSystemPersist.TEMPDIR_PREFIX + this._options.instrumentationKey
     );
 
-    fs.stat(tempDir, (statErr: Error | null, stats: fs.Stats) => {
-      if (statErr) {
-        if (((statErr as unknown) as { code: string }).code === "ENOENT") {
-          callback(null);
+    try {
+      const stats = await statAsync(tempDir);
+      if (stats.isDirectory()) {
+        const origFiles = await readdirAsync(tempDir);
+        const files = origFiles.filter((f) =>
+          path.basename(f).includes(FileSystemPersist.FILENAME_SUFFIX)
+        );
+        if (files.length === 0) {
+          return null;
         } else {
-          callback(statErr);
+          const firstFile = files[0];
+          const filePath = path.join(tempDir, firstFile);
+          const payload = await readFileAsync(filePath);
+          // delete the file first to prevent double sending
+          await unlinkAsync(filePath);
+          return payload;
         }
-      } else if (stats.isDirectory()) {
-        fs.readdir(tempDir, (error, origFiles) => {
-          if (!error) {
-            const files = origFiles.filter((f) =>
-              path.basename(f).includes(FileSystemPersist.FILENAME_SUFFIX)
-            );
-            if (files.length === 0) {
-              callback(null);
-            } else {
-              const firstFile = files[0];
-              const filePath = path.join(tempDir, firstFile);
-              fs.readFile(filePath, (readFileErr, payload) => {
-                if (!readFileErr) {
-                  // delete the file first to prevent double sending
-                  fs.unlink(filePath, (unlinkError) => {
-                    if (!unlinkError) {
-                      callback(null, payload);
-                    } else {
-                      callback(unlinkError);
-                    }
-                  });
-                } else {
-                  callback(readFileErr);
-                }
-              });
-            }
-          } else {
-            callback(error);
-          }
-        });
       }
-    });
+      return null;
+    } catch (e) {
+      if (((e as unknown) as { code: string }).code === "ENOENT") {
```

Comment:
since exception are always typed as `any`, the cast to unknown is not necessary, and I'm not sure if casting to `{code: string}` really helps anything. 

```suggestion
      if (e.code === "ENOENT") {
```

Seems fine here